openapi_schema_to_json_schema/to_jsonschema.py

```diff
@@ -43,12 +43,53 @@ def convert(schema, options=None):
     if options['cloneSchema']:
         schema = json.loads(json.dumps(schema))
 
+    return schema, options
+
+
+def convert(schema, options=None):
+    schema, options = _prepare(schema, options)
+
     schema = convertSchema(schema, options)
     schema['$schema'] = 'http://json-schema.org/draft-04/schema#'
 
     return schema
 
 
+def _recurse(tree, options):
+    for key, subtree in list(tree.items()):
+        if isinstance(subtree, dict):
+            if key == 'schema':
+                tree[key] = convertSchema(subtree, options)
+            else:
+                tree[key] = _recurse(subtree, options)
+        elif isinstance(subtree, list):
+            tree[key] = [
+                _recurse(item, options) if isinstance(item, dict) else item
+                for item in subtree
+            ]
+
+    return tree
+
+
+def convertDoc(doc, options):
+    doc, options = _prepare(doc, options)
+
+    components_schemas = doc.get('components', {}).get('schemas')
+    if components_schemas:
+        for name, struct in list(components_schemas.items()):
+            components_schemas[name] = convertSchema(struct, options)
+
+    paths = doc.get('paths')
+
+    if paths:
+        doc['paths'] = dict((path, _recurse(tree, options))
+                            for path, tree in paths.items())
+
+    doc['$schema'] = 'http://json-schema.org/draft-04/schema#'
```

Comment:
This line probably should be removed. The `$schema` field is only relevant for [JSON Schema](https://json-schema.org/understanding-json-schema/reference/schema.html), so it doesn't make sense at the top level of the resulting OpenAPI document (which is not processed by JSON Schema validators/tools).

That said, the resulting document is technically no longer valid OpenAPI, so it's fine if there's a reason for it.

Reply:
I was thinking of having some easy way to check whether the doc has already been transformed to using JSON Schemas.  It could be any identifier; open to suggestions on what might be appropriate.

Reply:
Perhaps adding an `$id` ?  https://json-schema.org/draft/2019-09/json-schema-core.html#rfc.section.8.2.2.1

Reply:
I think it's fine as-is.